lxd/main_init_interactive.go

```diff
@@ -495,6 +495,15 @@ func (c *cmdInit) askStoragePool(config *cmdInitData, d lxd.ContainerServer, poo
 
 				// Ask for the number of placement groups
 				pool.Config["ceph.osd.pg_num"] = cli.AskString("Number of placement groups [default=32]: ", "32", nil)
+			}
+			if pool.Driver == "zfs" {
```

Comment:
This check is at the wrong place, at this point, we have no idea what dataset the user intends to be using, they may still provide us the name of a completely different dataset, avoiding the conflict.

Reply:
And that check should also be restricted to when running as root.